pages/map.py

```diff
@@ -211,31 +272,48 @@ def update_user_ids_options(trips_data, selected_user_ids):
     Input('user-email-dropdown', 'value'),
 )
 def update_user_emails_options(trips_data, selected_user_emails):
-    user_emails_options, user_emails = create_user_emails_options(trips_data['users_data'])
+    user_emails_options, user_emails = create_user_emails_options(trips_data[0]['users_data'])
     if selected_user_emails is not None:
         selected_user_emails = [user_email for user_email in selected_user_emails if user_email in user_emails]
     return user_emails_options, selected_user_emails
 
+@callback(
+    Output('user-mode-dropdown', 'options'),
+    Output('user-mode-dropdown', 'value'),
+    Input('store-trips-map', 'data'),
+    Input('user-mode-dropdown', 'value'),
+)
+def update_user_modes_options(trips_data, selected_user_modes):
+    user_modes_options, user_modes = create_user_modes_options(trips_data[1]['users_data'])
+    if selected_user_modes is not None:
+        selected_user_modes = [mode_confirm for mode_confirm in selected_user_modes if mode_confirm in user_modes]
+    return user_modes_options, selected_user_modes
 
 @callback(
     Output('trip-map', 'figure'),
     Input('map-type-dropdown', 'value'),
     Input('user-id-dropdown', 'value'),
     Input('user-email-dropdown', 'value'),
+    Input('user-mode-dropdown', 'value'),
     State('store-trips-map', 'data'),
 )
-def update_output(map_type, selected_user_ids, selected_user_emails, trips_data):
+def update_output(map_type, selected_user_ids, selected_user_emails, selected_user_modes, trips_data):
     user_ids = set(selected_user_ids) if selected_user_ids is not None else set()
+    user_modes=set(selected_user_modes) if selected_user_modes is not None else set()
     if selected_user_emails is not None:
         for user_email in selected_user_emails:
             user_ids.add(str(ecwu.User.fromEmail(user_email).uuid))
-
+    arg1 = trips_data[0].get('users_data', {})
+    arg2 = user_ids       
+    if(selected_user_modes):
+        arg1 = trips_data[1].get('users_data', {})
+        arg2 = user_modes
```

Comment:
why are these variables called `arg1` and `arg2`?
If they do need to be called those names, why are they not used consistently?